### PR TITLE
Add new AP bandwidth API

### PR DESCRIPTION
**Pull request by nxf58150**
Add new API to configure AP bandwidth.

### CODE REVIEW

File: subsys/net/l2/wifi/wifi_shell.c [hunk 1]
```diff
@@ -1678,6 +1681,23 @@ static int wifi_ap_config_args_to_params(const struct shell *sh, size_t argc, ch
 			params->max_num_sta = (uint32_t)val;
 			params->type |= WIFI_AP_CONFIG_PARAM_MAX_NUM_STA;
 			break;
+		case 'I':
+			params->if_index = (uint8_t)atoi(optarg);
```
Comment (jukkar) [inline subsys/net/l2/wifi/wifi_shell.c:1685]: We are trying to migrate away from `atoi` calls as it does not check errors. There are still atoi calls in the shell but these will need to be converted in the future. Please use `shell_strtol()` call instead here.

File: subsys/net/l2/wifi/wifi_shell.c [hunk 2]
```diff
@@ -1678,6 +1681,23 @@ static int wifi_ap_config_args_to_params(const struct shell *sh, size_t argc, ch
...
+			break;
+		case 'b':
+			val = atoi(optarg);
```
Comment (jukkar) [inline subsys/net/l2/wifi/wifi_shell.c:1688]: Same here

File: subsys/net/l2/wifi/wifi_shell.c (hunk 1 above)
Comment (nxf58150) [reply subsys/net/l2/wifi/wifi_shell.c]: Updated.

File: subsys/net/l2/wifi/wifi_shell.c (hunk 2 above)
Comment (nxf58150) [reply subsys/net/l2/wifi/wifi_shell.c]: Updated.

**Review by jukkar — DISMISSED** [review]

**Comment by kartben** [comment]
@nxf58150 please rebase to address multiple merge conflicts. Thanks!

**Pull request closed without merging**